iceoryx2-ffi/cxx/include/iox2/active_request.hpp

```diff
@@ -139,29 +179,99 @@ template <ServiceType Service,
+    }
+    return iox::err(iox::into<LoanError>(result));
+}
+
+template <ServiceType Service,
+          typename RequestPayload,
+          typename RequestHeader,
+          typename ResponsePayload,
+          typename ResponseHeader>
+template <typename T, typename>
+inline auto ActiveRequest<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::loan_slice_uninit(
+    uint64_t number_of_elements)
+    -> iox::expected<ResponseMutUninit<Service, ResponsePayload, ResponseHeader>, LoanError> {
+    ResponseMutUninit<Service, ResponsePayload, ResponseHeader> response;
+    auto result = iox2_active_request_loan_slice_uninit(
+        &m_handle, &response.m_response.m_response, &response.m_response.m_handle, number_of_elements);
+    if (result == IOX2_OK) {
+        return iox::ok(std::move(response));
+    }
+    return iox::err(iox::into<LoanError>(result));
 }
 
 template <ServiceType Service,
           typename RequestPayload,
           typename RequestHeader,
           typename ResponsePayload,
           typename ResponseHeader>
+template <typename T, typename>
 inline auto ActiveRequest<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::send_copy(
-    [[maybe_unused]] const ResponsePayload& value) const -> iox::expected<void, SendError> {
-    IOX_TODO();
+    const ResponsePayload& payload) const -> iox::expected<void, SendError> {
+    static_assert(std::is_trivially_copyable_v<ResponsePayload>);
+
+    auto result =
+        iox2_active_request_send_copy(&m_handle, static_cast<const void*>(&payload), sizeof(ResponsePayload), 1);
+    if (result == IOX2_OK) {
+        return iox::ok();
+    }
+    return iox::err(iox::into<SendError>(result));
 }
 
 template <ServiceType Service,
           typename RequestPayload,
           typename RequestHeader,
           typename ResponsePayload,
           typename ResponseHeader>
+template <typename T, typename>
+inline auto ActiveRequest<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::send_slice_copy(
+    const iox::ImmutableSlice<ValueType>& payload) const -> iox::expected<void, SendError> {
+    auto result = iox2_active_request_send_copy(
+        &m_handle, payload.data(), sizeof(typename ResponsePayload::ValueType), payload.number_of_elements());
+    if (result == IOX2_OK) {
+        return iox::ok();
+    }
+    return iox::err(iox::into<SendError>(result));
+}
+
+template <ServiceType Service,
+          typename RequestPayload,
+          typename RequestHeader,
+          typename ResponsePayload,
+          typename ResponseHeader>
+template <typename T, typename>
 inline auto ActiveRequest<Service, RequestPayload, RequestHeader, ResponsePayload, ResponseHeader>::payload() const
-    -> const RequestPayload& {
-    IOX_TODO();
+    -> const T& {
+    const void* ptr = nullptr;
+    size_t number_of_elements = 0;
+
+    iox2_active_request_payload(&m_handle, &ptr, &number_of_elements);
```

Comment:
Could you remove `number_of_elements` and provide just a `nullptr` since you do not use it.